src/sisl/geom/neighs/_neigh_operations.py

```diff
@@ -325,7 +329,7 @@ def get_all_unique_pairs(
 
         for j in range(8):
             # Find the bin index.
-            bin_index = indices[at, j]
+            bin_index: cnp.int64_t = indices[at, j]
```

Comment:
Have you checked if this is necessary? I have found that cython 3 can determine the type of the variable if it is assigned from a typed array.

Reply:
mainly for clarity, agreed it shouldn't be required, perhaps this was when I used 0.29 it popped up. I'll do another annotation cycle of it.